pkg/rtc/mediatracksubscriptions.go

```diff
@@ -399,29 +399,23 @@ func (t *MediaTrackSubscriptions) downTrackClosed(
 	if !willBeResumed {
 		t.params.Telemetry.TrackUnsubscribed(context.Background(), subscriberID, t.params.MediaTrack.ToProto())
 
-		// if the source has been terminated, we'll need to terminate all the subscribed tracks
-		// however, if the dest sub has disconnected, then we can skip
-		if sender == nil {
```

Comment:
We want `RemoveSubscribedTrack` below to run even if this is the case (actually this should not happen at all). So, removing all the early returns and running remove from PC code only when appropriate